sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -1119,11 +1121,32 @@ class GroupedAggregations extends Arm {
       boundOrderSpec: Seq[SortOrder],
       orderByPositions: Array[Int],
       partByPositions: Array[Int],
-      inputCb: ColumnarBatch,
+      inputSpillable: SpillableColumnarBatch,
       outputColumns: Array[cudf.ColumnVector]): Unit = {
-    doRunningWindowOptimizedAggs(isRunningBatched, partByPositions, inputCb, outputColumns)
-    doRowAggs(boundOrderSpec, orderByPositions, partByPositions, inputCb, outputColumns)
-    doRangeAggs(boundOrderSpec, orderByPositions, partByPositions, inputCb, outputColumns)
+    withRetryNoSplit(inputSpillable) { attempt =>
+      // when there are exceptions in this body, we always want to close
+      // `outputColumns` before a likely retry.
+      try {
+        withResource(attempt.getColumnarBatch()) { attemptCb =>
+          doRunningWindowOptimizedAggs(
+            isRunningBatched, partByPositions, attemptCb, outputColumns)
+          doRowAggs(
+            boundOrderSpec, orderByPositions, partByPositions, attemptCb, outputColumns)
+          doRangeAggs(
+            boundOrderSpec, orderByPositions, partByPositions, attemptCb, outputColumns)
+        }
+      } catch {
+        case t: Throwable =>
+          // on exceptions we want to throw away any columns in outputColumns that
+          // are not pass-through
+          val columnsToClose = outputColumns.filter(_ != null)
```

Comment:
Is it worth making something like this a part of `withRetry`? and `withRetryNoSplit`?

Reply:
if we find we need this pattern in other places, yes, happy to add. Will keep it as is for now but agree, we could have some sort of "closer" that should be executed in case of exception.